### DISCUSSION TITLE
Unexpected token: operator (<) in terser plugin

### DISCUSSION THREAD

**Discussion by rastafrange** [facebook/docusaurus#6219] [Q&A] [answered]
Hi, 
I have updated to last version of docusaurus but when I try to `npm run build`I have the following error:

```
assets/files/index-6d39b751c123b72178cd08aade0bce19.js from Terser plugin
Unexpected token: operator (<) [assets/files/index-6d39b751c123b72178cd08aade0bce19.js:12,11]
[ERROR] Client bundle compiled with errors therefore further build is impossible.
```

My `package.json` file:

```
{
  "name": "rastavoyage",
  "version": "1.2.2",
  "private": true,
  "scripts": {
    "docusaurus": "docusaurus",
    "start": "docusaurus start",
    "build": "docusaurus build",
    "swizzle": "docusaurus swizzle",
    "deploy": "docusaurus deploy",
    "serve": "docusaurus serve"
  },
  "dependencies": {
    "@docusaurus/core": "^2.0.0-beta.14",
    "@docusaurus/plugin-google-analytics": "^2.0.0-beta.14",
    "@docusaurus/plugin-google-gtag": "^2.0.0-beta.14",
    "@docusaurus/plugin-ideal-image": "^2.0.0-beta.14",
    "@docusaurus/plugin-sitemap": "^2.0.0-beta.14",
    "@docusaurus/preset-classic": "^2.0.0-beta.14",
    "@hapi/hapi": "^20.0.2",
    "@material-ui/core": "^4.11.2",
    "@material-ui/icons": "^4.11.2",
    "@material-ui/lab": "^4.0.0-alpha.57",
    "@mdx-js/react": "^1.5.8",
    "@sideway/address": "^4.1.0",
    "@sideway/formula": "^3.0.0",
    "@sideway/pinpoint": "^2.0.0",
    "autoprefixer": "9.0.0",
    "clsx": "^1.1.1",
    "commentbox.io": "^2.0.0",
    "core-js": "^3.20.1",
    "emailjs-com": "^2.6.4",
    "formik": "^2.2.5",
    "immer": "^9.0.7",
    "joi": "^17.3.0",
    "postcss": "^8.3.2",
    "postcss-loader": "^4.0.2",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-google-recaptcha": "^2.1.0",
    "react-hook-form": "^6.12.2",
    "react-toastify": "^6.1.0",
    "swal": "^0.1.0",
    "sweetalert2": "^10.11.1"
    },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.16.5",
    "@babel/preset-env": "^7.16.5",
    "@babel/preset-react": "^7.16.5",
    "@docusaurus/module-type-aliases": "^2.0.0-beta.14",
    "@tsconfig/docusaurus": "^1.0.2",
    "@types/react": "^16.9.56",
    "@types/react-helmet": "^6.1.0",
    "@types/react-router-dom": "^5.1.6",
    "babel-loader": "^8.2.3",
    "typescript": "^4.0.5",
    "webpack": "^5.65.0"
  }
}

```

[My repo](https://github.com/rastafrange/blog) if it can help

I feel like I have tried everything, I just don't understand this error (I don't even know what file it is talking about)

What I have tried:

- installing previous versions of Terser plugin
- [this](https://stackoverflow.com/questions/53957787/reactjs-unexpected-token?noredirect=1&lq=1) : maybe I haven't done this well but has not succeeded

Thanks

**Comment by slorber** [comment]
Hi,

I don't know what is causing this problem unfortunately

The repro is not beta.14, that would help if you try to upgrade the repo to the latest version (can be a separate branch) so that we can be sure to test exactly the same code as yours

You can also run the `docusaurus clear` command, and try to regenerate your package manager lockfile

**Reply by rastafrange** [reply]
Just push on [another branch](https://github.com/rastafrange/blog/tree/beta.14) to facilitate debug. 
I already ran `docusaurus clear` many times, removing node_modules/* package-lock.json etc and it has not worked. 
Thank you for your help :)

**Comment by slorber** [comment, marked as answer]
This issue has been solved.

The problem was using a markdown link linking to a JS file, not sure why but it produced this weird behavior.

`[carnets](../../../../src/pages/carnets/index.js)`

**Reply by Josh-Cena** [reply]
My guess is that the default loader still kicks in even when we use `file-loader` (see also #4827). Need a way to chain a `raw-loader` first

**Comment by Alan4247748** [comment]
Two years later error still exists.

`ERROR:`
_assets/files/index-5514bd03c280a30bcc18627702046ed7.js from Terser plugin
Unexpected token: operator (<) [assets/files/index-5514bd03c280a30bcc18627702046ed7.js:13,4]_

`FIX: `
_Don't use a direct link from a .md file to a .js file - use the (URL link)._ 

So strange - no idea - why it gets caused. I checked the plugin source code—no help. 
And I never could find where the h*ll [assets/files/index-5514bd03c280a30bcc18627702046ed7.js:13,4] lives??

**Reply by rastafrange** [reply]
Hello, are you sure you don't have the character '<' anywhere? In the end, it was in one of my json file if I remember correctly? However it is true that the errors are very confusing and do not indicate exactly the source of the errors (which file)

**Reply by Alan4247748** [reply]
Hey, yes, I'm sure there was no '<' anywhere. I tested it on several pages. A direct (in-page) link from a .md file to a .js file — always produced the unexpected token error.
